repub_test.go

```diff
@@ -14,12 +14,14 @@ func TestRepublisher(t *testing.T) {
 		t.Skip("dont run timing tests in CI")
 	}
 
-	ctx := context.TODO()
-
 	pub := make(chan struct{})
 
 	pf := func(ctx context.Context, c cid.Cid) error {
-		pub <- struct{}{}
+		select {
+		case pub <- struct{}{}:
+		case <-ctx.Done():
+			return ctx.Err()
+		}
```

Comment:
This, along with the way `ctx` is passed as an argument now instead of storing it in the `Republisher`, makes much more sense.